### PR TITLE
Adding a BackingField attribute

### LINKED ISSUE #6266: Provide a data annotation for setting backing fields
This would allow a property to be annotated with a backing field directly in the class. One advantage of this is that the nameof() syntax could be used to reference the field.

### PR DESCRIPTION
**Pull request by lajones**
Fixes #6266. Adding a `BackingField` attribute which does the same as the `HasField()` fluent API.

### CODE REVIEW

File: src/EFCore.Abstractions/BackingFieldAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    /// <summary>
+    ///     Names the backing field associated with this property.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property)]
+    public sealed class BackingFieldAttribute : Attribute
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="BackingFieldAttribute" /> class.
+        /// </summary>
+        /// <param name="name">The name of the backing field.</param>
```
Comment (smitpatel) [inline src/EFCore.Abstractions/BackingFieldAttribute.cs]: space between tag & text.

File: src/EFCore/Metadata/Conventions/Infrastructure/ProviderConventionSetBuilder.cs [hunk 2]
```diff
@@ -210,6 +212,7 @@ public virtual ConventionSet CreateConventionSet()
             conventionSet.PropertyFieldChangedConventions.Add(maxLengthAttributeConvention);
             conventionSet.PropertyFieldChangedConventions.Add(stringLengthAttributeConvention);
             conventionSet.PropertyFieldChangedConventions.Add(timestampAttributeConvention);
+            conventionSet.PropertyFieldChangedConventions.Add(backingFieldAttributeConvention);
```
Comment (AndriySvyryd) [inline src/EFCore/Metadata/Conventions/Infrastructure/ProviderConventionSetBuilder.cs]: No need to add it here, the convention doesn't react to the backing field being changed

File: src/EFCore.Abstractions/BackingFieldAttribute.cs (hunk 1 above)
Comment (lajones) [reply src/EFCore.Abstractions/BackingFieldAttribute.cs]: Will fix

File: src/EFCore/Metadata/Conventions/Infrastructure/ProviderConventionSetBuilder.cs (hunk 2 above)
Comment (lajones) [reply src/EFCore/Metadata/Conventions/Infrastructure/ProviderConventionSetBuilder.cs]: Will fix.

**Pull request merged**